### PR TITLE
[mapbox-gl] Add missing properties for layer types

### PR DESCRIPTION
**Pull request by amxfonseca**
The main intention of this PR was to add some missing properties for some of the layout definitions (`FillLayout`, `LineLayout`, `SymbolLayout` and `CircleLayout`). In meanwhile there is also some other minor stuff bundled with this PR, I will list it bellow:

1) ~~The code formatting had some inconsistencies. So I decided to pass it through prettier, since it is a kind of accepted formatting standard these days.~~ See #43443

2) Created a new named type `AnyPaint` that contains all the paint layer definitions. There is already a `AnyLayout` in place, so it makes sense to also have the same behavior for paint.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
    + Tested it on a big Angular app that I am working on, everything compiles properly
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
    + https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-fill-fill-sort-key
    + https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-line-line-sort-key
    + https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-symbol-sort-key
    + https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-text-radial-offset
    + https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-text-variable-anchor
    + https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-text-writing-mode
    + https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-circle-circle-sort-key

### CODE REVIEW

**Comment by dmytro-gokun** [comment]
@amxfonseca Thanks a lot for the PR! It looks great. I would prefer though that this part: "I decided to pass it through prettier, since it is a kind of accepted formatting standard these days" was not done or done via a separate PR. Also, when running prettier.. did you use settings from this repo as described here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request or?

**Comment by amxfonseca** [comment]
You are right. I was a bit lazy I confess 🙂

I'll update this PR later and create a separate one for the formatting changes 👍

**Comment by amxfonseca** [comment]
Hey @dmytro-gokun!

Finally updated the PR. No more code formatting voodoo going on 😄.  I've done that on a separate PR though (#43443) since we want to use prettier in the future anyway. 

I think it's best if we merge this one first, and I can properly rebase the other one to make sure the final formatting is right 👍

File: types/mapbox-gl/index.d.ts [hunk 1]
```diff
@@ -1796,6 +1810,10 @@ declare namespace mapboxgl {
         'text-allow-overlap'?: boolean;
         'text-ignore-placement'?: boolean;
         'text-optional'?: boolean;
+        'text-radial-offset'?: number | Expression;
+        'text-variable-anchor'?: Anchor[];
+        'text-writing-mode'?: ['horizontal', 'vertical'];
```
Comment (dmytro-gokun) [inline types/mapbox-gl/index.d.ts]: I'm not sure about this one should not it be ('horizontal' | 'vertical')[] instead?

File: types/mapbox-gl/mapbox-gl-tests.ts [hunk 2]
```diff
@@ -1098,3 +1098,9 @@ expectType<mapboxgl.Visibility>('none');
 expectType<mapboxgl.AnyLayout>({visibility: 'none'});
 expectType<mapboxgl.AnyLayout>({visibility: 'none', 'line-cap': 'round' });
 expectType<mapboxgl.AnyLayout>({visibility: 'visible', 'icon-allow-overlap': true });
+
+/*
+ * AnyPaint
+ */
+expectType<mapboxgl.AnyPaint>({ 'circle-pitch-alignment': 'map' });
```
Comment (dmytro-gokun) [inline types/mapbox-gl/mapbox-gl-tests.ts]: Do you mind creating more test cases so all new definitions are covered?

Comment (amxfonseca) [reply types/mapbox-gl/mapbox-gl-tests.ts]: Yes! The tests are a bit lacking. There wasn’t basically any test in place for most layers. I can improve that.

I’ll add tests for all the existing layers.

File: types/mapbox-gl/index.d.ts (hunk 1 above)
Comment (amxfonseca) [reply types/mapbox-gl/index.d.ts]: I really missed that one. I’ll fix it 👍

**Comment by amxfonseca** [comment]
Just updated the PR. Since there weren't any tests in place for layout and pain options I decided to add new tests for every one of them.

They are fairly thorough in the way that every property is tested with all the possible types.

**Review by dmytro-gokun — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**